### PR TITLE
[JENKINS-35993] Fix Blue Ocean button for when no context root

### PR DESCRIPTION
**Pull request by tfennelly**
Related to issue [JENKINS-35993](https://issues.jenkins-ci.org/browse/JENKINS-35993). 

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

File: blueocean-web/src/main/js/try.js [hunk 1]
```diff
@@ -1,6 +1,19 @@
 var $ = require('jquery-detached').getJQuery();
 var jsModules = require('@jenkins-cd/js-modules');
 
+function getAppUrl() {
+    var rootUrl = jsModules.getRootURL();
+    
+    // If there's no prefix, rooturl is blank, resulting in js-modules
+    // returning the static url. We don't want to use that when
+    // switching to blue ocean, so return '' instead.
+    if (rootUrl.indexOf('/static/') === 0) {
```
Comment (kzantow) [inline blueocean-web/src/main/js/try.js]: why would this ever be?

**Comment by kzantow** [comment]
:bee: still wish for this just being an <a href...

File: blueocean-web/src/main/js/try.js (hunk 1 above)
Comment (tfennelly) [inline blueocean-web/src/main/js/try.js]: Read the comment.

Comment (kzantow) [inline blueocean-web/src/main/js/try.js]: ... I mean, why would js-modules ever return the static URL as the root URL? this isn't normal behavior. handling it here is the wrong place, fix js-modules to not do this... if you need that static resource url, create a method for it if there isn't one.

Comment (tfennelly) [inline blueocean-web/src/main/js/try.js]: @kzantow We don't actually want the static url ... js-modules just falls back to giving that when the root url attribute is not set for some reason e.g. when there's no context path, or on older versions of Jenkins that just don't set it at all. This is ok for what js-modules does, but I'll just remove it and access the DOM directly. Gimme a min 😄

**Comment by kzantow** [comment]
🐝  seems better

**Comment by kzantow** [comment]
stilla :bee:

**Pull request merged**